src/lib.rs

```diff
@@ -0,0 +1,400 @@
+
+    let package: Package = parser::parse(xml).expect("Failed to parse XML!");
+    let document: Document = package.as_document();
+
+    let xml_element = get_xml_element(document);
+
+    for child in xml_element.children().iter() {
+        match child {
+            &ChildOfElement::Element(el) => {
+                let element_name = el.name().local_part();
+                match element_name {
+                    "block" => {
+                        program.groups.push(make_statement_body(el));
+                    },
+                    _ => {}
+                }
+            },
+            _ => {}
+        }
+    }
+
+    program
+}
+
+fn get_next_block_element(block_el: Element) -> Option<Element> {
+    let next_el: Option<Element> = {
+        let mut found: Option<Element> = None;
+        for child in block_el.children().iter() {
+            match child {
+                &ChildOfElement::Element(el) => {
+                    match el.name().local_part() {
+                        "next" => {
+                            found = Some(el);
+                            break;
+                        },
+                        _ => {}
+                    }
+                },
+                _ => {}
+            }
+        }
+        found
+    };
+
+    match next_el {
+        Some(next_el) => {
+            for child in next_el.children().iter() {
+                match child {
+                    &ChildOfElement::Element(el) => {
+                        match el.name().local_part() {
+                            "block" => {
+                                return Some(el);
+                            },
+                            _ => {}
+                        }
+                    },
+                    _ => {}
+                }
+            }
+            None
+        },
+        None => None
+    }
+}
+
+fn make_statement_body(first_block: Element) -> StatementBody {
+    let mut body = StatementBody::new();
+
+    // Create each block, put them into the statement body
+    let mut block_el: Element;
+    block_el = first_block;
+    loop {
+        body.blocks.push(make_block(block_el));
+        let next_block = get_next_block_element(block_el);
+        match next_block {
+            Some(next_block_el) => {
+                block_el = next_block_el;
+            },
+            None => {
+                break;
+            }
+        }
+    }
+
+    body
+}
+
+fn make_block(block_el: Element) -> Block {
+    let mut block = Block::new();
+    for attribute in block_el.attributes().iter() {
+        let name = attribute.name().local_part();
+        let value = attribute.value().to_string();
+        match name {
+            "type" => { block.block_type = value; },
+            "id" => { block.id = value; },
+            _ => {}
+        }
+    }
+    for child in block_el.children().iter() {
+        match child {
+            &ChildOfElement::Element(child_el) => {
```

Comment:
You have multiple of this `match foobar { Foo::Bar => { ... }, _ {} }`. Notice that if you use `match` for a single case, where all other cases are "do nothing", a `if let` is more concise. A `match` prepares the reader to expect more than one case, a `if let` is "this is the pattern, if else do not care about this intended stuff".

Reply:
@lukaslueg You're right! I had forgotten about `if let` but that seems appropriate in several places. Thanks!